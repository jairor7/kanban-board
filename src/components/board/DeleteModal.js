import React from "react";
import { connect } from "react-redux";
import Modal from "react-modal";

Modal.setAppElement("#root");

export const DeleteModal = ({
  activeDeleteTask,
  handleDeleteModal,
  deleteTask,
}) => {
  return (
    <Modal isOpen={activeDeleteTask}>
      <h1>Eliminar tarea</h1>
      <p>Esta seguro que desea eliminar esta tarea</p>
      <button onClick={handleDeleteModal}>Cancelar</button>
      <button onClick={deleteTask}>Eliminar</button>
    </Modal>
  );
};

const mapStateToProps = (state) => ({});

const mapDispatchToProps = {};

export default connect(mapStateToProps, mapDispatchToProps)(DeleteModal);
