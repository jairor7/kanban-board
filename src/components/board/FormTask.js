import React from "react";
import { connect } from "react-redux";

export const FormTask = ({
  handleTask,
  handleDescription,
  handlePriority,
  handleStateTask,
  handleHours,
  task,
  description,
  priority,
  stateTask,
  hours,
  isEdit,
}) => {
  return (
    <div className="form-container row-3 col-2">
      <div className="form-group">
        <label className="required">Título:</label>
        <input
          type="text"
          maxLength="50"
          id="task"
          placeholder="Nombre de la tarea"
          onChange={handleTask}
          defaultValue={task}
          required="required"
          autoComplete="off"
        />
      </div>

      <div className="form-group">
        <label>Tiempo (hr):</label>
        <input
          type="text"
          maxLength="5"
          id="hours"
          onChange={handleHours}
          defaultValue={hours}
        />
      </div>
      <div className="form-group input-description">
        <label>Descripción:</label>
        <textarea
          id="description"
          placeholder="Descripción de la tarea"
          rows="3"
          maxLength="500"
          onChange={handleDescription}
          defaultValue={description}
        ></textarea>
      </div>
      <div className="form-group">
        <label>Prioridad:</label>
        <select id="priority" onChange={handlePriority} defaultValue={priority}>
          <option value={1}>Baja</option>
          <option value={2}>Media</option>
          <option value={3}>Alta</option>
        </select>
      </div>
      {isEdit && (
        <div className="form-group">
          <label>Estado:</label>
          <select
            id="state"
            onChange={handleStateTask}
            defaultValue={stateTask}
          >
            <option value={0}>Backlog</option>
            <option value={1}>To Do</option>
            <option value={2}>Doing</option>
            <option value={3}>Code Review</option>
            <option value={4}>Done</option>
          </select>
        </div>
      )}
    </div>
  );
};

const mapStateToProps = (state) => ({});

const mapDispatchToProps = {};

export default connect(mapStateToProps, mapDispatchToProps)(FormTask);
