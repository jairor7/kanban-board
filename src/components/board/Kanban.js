import React from "react";
import { DragDropContext, Droppable, Draggable } from "react-beautiful-dnd";
import { connect } from "react-redux";
import { selectTasksByState } from "../../store/redux/selectors/tasks";
import { stateTaskName, stateTask } from "../../utils/constants";
import { editTask } from "../../store/redux/actions/kanbanAction";
import TaskCard from "../general/TaskCard";

export const Kanban = (props) => {
  const onDragEnd = (result) => {
    const { destination, source, draggableId } = result;
    if (!destination) {
      return;
    } else if (
      source.index === destination.index &&
      source.droppableId === destination.droppableId
    ) {
      return;
    }
    let task = props.allTask.find((task) => task.id === draggableId);
    task.stateTask = stateTaskName.indexOf(destination.droppableId);
    props.editTask(props.uid, task, draggableId);
  };

  return props.totalTask > 0 ? (
    <div className="container container-dashboard">
      {props.orderedTasks && (
        <DragDropContext onDragEnd={(result) => onDragEnd(result)}>
          {props.orderedTasks.map((tasks, index) => (
            <div key={index} className="container-task-state">
              <h2>
                {stateTaskName[index]} ({tasks.length})
              </h2>
              <hr />
              <Droppable
                droppableId={stateTaskName[index]}
                className="container box-kanban"
              >
                {(provided) => (
                  <ul
                    {...provided.droppableProps}
                    ref={provided.innerRef}
                    className="box-kanban"
                  >
                    {tasks.map((task, index) => (
                      <Draggable
                        key={task.id}
                        draggableId={task.id}
                        index={index}
                      >
                        {(draggableProvider) => (
                          <li
                            className={`task-card-${task.priority}`}
                            ref={draggableProvider.innerRef}
                            {...draggableProvider.draggableProps}
                            {...draggableProvider.dragHandleProps}
                          >
                            <TaskCard task={task} />
                          </li>
                        )}
                      </Draggable>
                    ))}
                    {provided.placeholder}
                  </ul>
                )}
              </Droppable>
            </div>
          ))}
        </DragDropContext>
      )}
    </div>
  ) : (
    <div>No hay datos para mostrar</div>
  );
};

const mapStateToProps = (state) => ({
  orderedTasks: selectTasksByState(state.kanbanReducer.tasks),
  totalTask: state.kanbanReducer.tasks.length,
  allTask: state.kanbanReducer.tasks,
  uid: state.loginReducer.user.uid,
});

const mapDispatchToProps = (dispatch) => ({
  editTask: (uid, task, taskId, history) =>
    dispatch(editTask(uid, task, taskId, history)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Kanban);
