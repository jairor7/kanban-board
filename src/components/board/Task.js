import React from "react";
import { connect } from "react-redux";

export const Task = (props) => {
  return <div>Task</div>;
};

const mapStateToProps = (state) => ({});

const mapDispatchToProps = {};

export default connect(mapStateToProps, mapDispatchToProps)(Task);
