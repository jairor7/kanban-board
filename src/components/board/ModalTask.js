import React, { useState, useEffect } from "react";
import Modal from "react-modal";
import { connect } from "react-redux";
import { history } from "../../routes/AppRouter";
import { addTask, editTask } from "../../store/redux/actions/kanbanAction";
import FormTask from "./FormTask";

Modal.setAppElement("#root");

export const ModalTask = ({
  activeTaskModal,
  setActiveTaskModal,
  isEdit,
  addTask,
  editTask,
  currentTask,
  ...props
}) => {
  const [task, setTask] = useState("");
  const [description, setDescription] = useState("");
  const [priority, setPriority] = useState(1);
  const [stateTask, setStateTask] = useState(0);
  const [hours, setHours] = useState("0");

  useEffect(() => {
    if (isEdit) {
      setTask(currentTask.task);
      setDescription(currentTask.description);
      setPriority(currentTask.priority);
      setStateTask(currentTask.stateTask);
      setHours(currentTask.hours);
    }
  }, [currentTask, isEdit]);

  const handleModalTask = (e) => {
    e.preventDefault();
    setActiveTaskModal((activeTaskModal) => !activeTaskModal);
  };

  const handleTask = (e) => {
    setTask(e.target.value);
  };

  const handleDescription = (e) => {
    setDescription(e.target.value);
  };

  const handlePriority = (e) => {
    setPriority(parseInt(e.target.value));
  };

  const handleStateTask = (e) => {
    setStateTask(parseInt(e.target.value));
  };

  const handleHours = (e) => {
    setHours(e.target.value);
  };

  const onSubmit = (e) => {
    e.preventDefault();
    const taskData = {
      task,
      description,
      priority,
      stateTask,
      hours,
    };
    if (isEdit) {
      editTask(props.uid, taskData, currentTask.id, history);
    } else {
      addTask(props.uid, taskData, history);
    }
    setActiveTaskModal(false);
  };

  const text = isEdit ? "Editar " : "Agregar";

  return (
    <Modal isOpen={activeTaskModal}>
      <h1>{text} tarea</h1>
      <form onSubmit={onSubmit}>
        <FormTask
          handleTask={handleTask}
          handleDescription={handleDescription}
          handlePriority={handlePriority}
          handleStateTask={handleStateTask}
          handleHours={handleHours}
          task={task}
          description={description}
          priority={priority}
          stateTask={stateTask}
          hours={hours}
          isEdit={isEdit}
        />
        <button onClick={handleModalTask}>Cancelar</button>
        <button>{text}</button>
      </form>
    </Modal>
  );
};

const mapStateToProps = (state) => ({
  uid: state.loginReducer.user.uid,
});

const mapDispatchToProps = (dispatch) => ({
  addTask: (uid, task, history) => dispatch(addTask(uid, task, history)),
  editTask: (uid, task, taskId, history) =>
    dispatch(editTask(uid, task, taskId, history)),
});

export default connect(mapStateToProps, mapDispatchToProps)(ModalTask);
