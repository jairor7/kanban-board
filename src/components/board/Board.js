import React, { useState } from "react";
import { connect } from "react-redux";
import Kanban from "./Kanban";
import ModalTask from "./ModalTask";

export const Board = (props) => {
  const [activeTaskModal, setActiveTaskModal] = useState(false);

  const handleModalTask = () => {
    setActiveTaskModal((activeTaskModal) => !activeTaskModal);
  };

  return (
    <section className="main-container">
      {activeTaskModal && (
        <ModalTask
          activeTaskModal={activeTaskModal}
          setActiveTaskModal={setActiveTaskModal}
          isEdit={false}
        />
      )}
      <div className="header-board">
        <div className="title-board">
          <span className="subtitle">Board</span>
          <h1 className="title">Hola {props.displayName}!</h1>
        </div>
        <button className="btn btn-primary" onClick={handleModalTask}>
          <div className="btn-content">
            <span className="content">Nueva tarea</span>
            <svg
              xmlns="http://www.w3.org/2000/svg"
              className="icon icon-add"
              fill="none"
              viewBox="0 0 24 24"
              stroke="currentColor"
            >
              <path
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={2}
                d="M12 4v16m8-8H4"
              />
            </svg>
          </div>
        </button>
      </div>
      <Kanban />
    </section>
  );
};

const mapStateToProps = (state) => ({
  userName: state.loginReducer.user,
  displayName: state.loginReducer.user.displayName,
});

const mapDispatchToProps = (dispatch) => ({});

export default connect(mapStateToProps, mapDispatchToProps)(Board);
