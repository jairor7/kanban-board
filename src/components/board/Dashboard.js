import React, { useState, useEffect } from "react";
import { connect } from "react-redux";
import { getAllTask } from "../../store/redux/actions/kanbanAction";
import Kanban from "./Kanban";
import ModalTask from "./ModalTask";

export const Dashboard = (props) => {
  const [activeTaskModal, setActiveTaskModal] = useState(false);

  useEffect(() => {
    props.getAllTask(props.uid);
  }, [props]);

  const handleModalTask = () => {
    setActiveTaskModal((activeTaskModal) => !activeTaskModal);
  };

  return (
    <div>
      {activeTaskModal && (
        <ModalTask
          activeTaskModal={activeTaskModal}
          setActiveTaskModal={setActiveTaskModal}
          isEdit={false}
        />
      )}
      Dashboard
      <div>
        <button onClick={handleModalTask}>Agregar Tarea</button>
      </div>
      <Kanban />
    </div>
  );
};

const mapStateToProps = (state) => ({
  uid: state.loginReducer.user.uid,
});

const mapDispatchToProps = (dispatch) => ({
  getAllTask: (uid) => dispatch(getAllTask(uid)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Dashboard);
