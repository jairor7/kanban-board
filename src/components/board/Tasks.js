import React, { useState } from "react";
import { connect } from "react-redux";
import TaskCard from "../general/TaskCard";
import ModalTask from "./ModalTask";

export const Tasks = ({ allTasks }) => {
  const [activeTaskModal, setActiveTaskModal] = useState(false);

  const handleModalTask = () => {
    setActiveTaskModal((activeTaskModal) => !activeTaskModal);
  };
  return (
    <section className="main-container">
      {activeTaskModal && (
        <ModalTask
          activeTaskModal={activeTaskModal}
          setActiveTaskModal={setActiveTaskModal}
          isEdit={false}
        />
      )}
      <div className="header-board">
        <div className="title-board">
          <span className="subtitle">Tareas</span>
          <h1 className="title">Vista general</h1>
        </div>
        <button className="btn btn-primary" onClick={handleModalTask}>
          <div className="btn-content">
            <span className="content">Nueva tarea</span>
            <svg
              xmlns="http://www.w3.org/2000/svg"
              className="icon icon-add"
              fill="none"
              viewBox="0 0 24 24"
              stroke="currentColor"
            >
              <path
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={2}
                d="M12 4v16m8-8H4"
              />
            </svg>
          </div>
        </button>
      </div>
      {allTasks.length > 0 ? (
        <ul className="box-kanban grid">
          {allTasks.map((task, index) => (
            <li key={index} className={`task-card-${task.priority}`}>
              <TaskCard task={task} />
            </li>
          ))}
        </ul>
      ) : (
        <div>
          <p>No hay tareas, agrega una para comenzar</p>
        </div>
      )}
    </section>
  );
};

const mapStateToProps = (state) => ({
  userName: state.loginReducer.user,
  displayName: state.loginReducer.user.displayName,
  allTasks: state.kanbanReducer.tasks,
});

const mapDispatchToProps = (dispatch) => ({});

export default connect(mapStateToProps, mapDispatchToProps)(Tasks);
