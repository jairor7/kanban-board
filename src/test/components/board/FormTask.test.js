import { FormTask } from "../../../components/board/FormTask";
import React from "react";
import toJSON from "enzyme-to-json";
import { shallow } from "enzyme";

test("should render FormTask correctly", () => {
  const wrapper = shallow(<FormTask />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});
