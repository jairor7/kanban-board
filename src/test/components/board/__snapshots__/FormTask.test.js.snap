// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FormTask correctly 1`] = `
<div
  className="form-container"
>
  <div
    className="form-group"
  >
    <label
      className="required"
    >
      Título:
    </label>
    <input
      autoComplete="off"
      className="form-control"
      id="task"
      maxLength="50"
      placeholder="Nombre de la tarea"
      required="required"
      type="text"
    />
  </div>
  <div
    className="form-group"
  >
    <label>
      Tiempo (hr):
    </label>
    <input
      className="form-control"
      id="hours"
      maxLength="5"
      type="text"
    />
  </div>
  <div
    className="form-group input-description"
  >
    <label>
      Descripción:
    </label>
    <textarea
      className="form-control"
      id="description"
      maxLength="500"
      placeholder="Descripción de la tarea"
      rows="3"
    />
  </div>
  <div
    className="form-group"
  >
    <label>
      Prioridad:
    </label>
    <select
      className="form-control"
      id="priority"
    >
      <option
        value={1}
      >
        Baja
      </option>
      <option
        value={2}
      >
        Media
      </option>
      <option
        value={3}
      >
        Alta
      </option>
    </select>
  </div>
</div>
`;
